clarifai/client/model.py

```diff
@@ -802,9 +850,17 @@ def generate_by_url(self,
         >>> model = Model("url") # Example URL: https://clarifai.com/clarifai/main/models/general-image-recognition
                     or
         >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
-        >>> stream_response = model.generate_by_url('url', 'image', deployment_id='deployment_id')
+        >>> stream_response = model.generate_by_url('url', deployment_id='deployment_id')
         >>> list_stream_response = [response for response in stream_response]
     """
+    if not input_type:
+      model_input_types = self.get_model_input_types()
+      if len(model_input_types) > 1:
+        raise UserError(
+            "Model has multiple input types. Please use model.predict() for this multi-modal model."
+        )
+      input_type = model_input_types[0]
+
```

Comment:
I would suggest may be having this in a internal function(`def _check_input_type()` or something) and reuse that since this was used in multiple functions.

Reply:
Done